### PR TITLE
refactor: split out a test util module

### PR DESCRIPTION
**Pull request by ckipp01**
In the Bloop code base the integration tests also actually use this
trait. It's only a single file so in reality we _could_ duplicate it in
Bloop, but I figured we could also simply publish it as a mini-module
that Bloop could just use in its integration tests. It's not really
cleaned up or anything special, so I'm unsure if anything other than
Bloop will actually use this.

Bloop would then simply add:

```scala
Dependencies.bloopConfigTestUtil % Test
```

To the plugin modules.

Refs: https://github.com/scalacenter/bloop/pull/1868

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -19,15 +19,20 @@ val scala213 = "2.13.10"
...
-val configName = "config"
+object Platforms {
+  val jvm = "jvm"
+  val js = "js"
+}
 
-trait CommonPublish extends CiReleaseModule with Mima {
+// NOTE: Once we publish configTestUtil we can move this down and mix it in with CommonPublish
+trait MimaCheck extends Mima {
+  override def mimaPreviousVersions = Seq("1.5.4")
```
Comment (tgodzik) [inline build.sc:30]: Do we have to update it manually after each release? Can it be automated?

**Review by tgodzik — COMMENTED** [review]
I think it should be fine to keep it in Bloop only. It's a lot of internal logic, so it would be akward to use anywhere else.

File: build.sc (hunk 1 above)
Comment (ckipp01) [reply build.sc:30]: I'll dig in a bit more, but the other Mill projects I've looked at all had it explicit. And by default this is an empty Seq so I believe you need to manually do it.

**Comment by ckipp01** [comment]
> I think it should be fine to keep it in Bloop only. It's a lot of internal logic, so it would be akward to use anywhere else.

Can you clarify what you mean? It'll only be Bloop that uses this right? But I'd still need to publish this though as both Bloop and this repo use it for testing.

**Comment by tgodzik** [comment]
> > I think it should be fine to keep it in Bloop only. It's a lot of internal logic, so it would be akward to use anywhere else.
> 
> Can you clarify what you mean? It'll only be Bloop that uses this right? But I'd still need to publish this though as both Bloop and this repo use it for testing.

Ach, ok! I didn't know it's also used here. I thought it's only used for testing gradle and maven integration.

**Comment by tgodzik** [comment]
Wait, but is it really? I only see the definition https://github.com/scalacenter/bloop-config/search?q=BaseConfigSuite

Or are talking about another class?

**Comment by ckipp01** [comment]
> Wait, but is it really? I only see the definition https://github.com/scalacenter/bloop-config/search?q=BaseConfigSuite

Whoa, you're right, how the heck did I miss this. I'll move this out of here, sorry for the confusion!

**Pull request closed without merging**